spec/requests/donations_requests_spec.rb

```diff
@@ -172,5 +172,95 @@
+        extra_item_name = "Extra Item"
+        extra_item = create(:item, organization: organization, name: extra_item_name)
+        item_to_delete_name = "Item To Delete"
+        item_to_delete = create(:item, organization: organization, name: item_to_delete_name)
+        TestInventory.create_inventory(organization, { storage_location.id => { item.id => 0, extra_item.id => 1 } })
+        original_quantity = 100
+        original_source = Donation::SOURCES[:manufacturer]
+        original_date = DateTime.new(2024)
+        donation = create(:manufacturer_donation, issued_at: original_date, organization: organization, storage_location: storage_location, source: original_source)
+        add_item_donation = donation.as_json.merge({
+          line_items_attributes: {
+            "0": { item_id: item.id, quantity: original_quantity },
+            "1": { item_id: item_to_delete.id, quantity: 1 }
+          }
+        })
+
+        put donation_path(id: donation.id, donation: add_item_donation)
+
+        distribution = {
+          storage_location_id: storage_location.id,
+          partner_id: create(:partner).id,
+          delivery_method: :delivery,
+          line_items_attributes: {
+            "0": { item_id: item.id, quantity: 90 }
+          }
+        }
+
+        post distributions_path(distribution: distribution, format: :turbo_stream)
+
+        edited_source = Donation::SOURCES[:product_drive]
+        edited_source_drive_name = "Test Product Drive"
+        edited_source_drive = create(:product_drive, name: edited_source_drive_name, organization: organization)
+        edited_source_drive_participant_business_name = "Test Business Name"
+        edited_source_drive_participant = create(:product_drive_participant, business_name: edited_source_drive_participant_business_name, organization: organization)
+        edited_storage_location_name = "Test Storage"
+        edited_storage_location = create(:storage_location, name: edited_storage_location_name, organization: organization)
+        edited_money = 10.0
+        edited_comment = "New test comment"
+        edited_date = "2019-01-01"
+        extra_quantity = 1
+        edited_quantity = 1
+
+        # deleted item is removed from line_items_attributes entirely
+        edited_donation = {
+          source: edited_source,
+          product_drive_id: edited_source_drive.id,
+          product_drive_participant_id: edited_source_drive_participant.id,
+          storage_location_id: edited_storage_location.id,
+          money_raised_in_dollars: edited_money,
+          comment: edited_comment,
+          issued_at: edited_date,
+          line_items_attributes: {
+            "0": { item_id: item.id, quantity: edited_quantity },
+            "1": { item_id: extra_item.id, quantity: extra_quantity }
+          }
+        }
+
+        put donation_path(id: donation.id, donation: edited_donation)
```

Comment:
This is really what we're trying to hit, right? Why are we hitting all those other endpoints rather than just directly creating the data we want to interact with on this page?

Reply:
I think when I wrote this (several weeks ago at this point, I don't remember) I didn't know how the inventory system worked and that DistributionCreateService or TestInventory existed. I think this can be rewritten to just use TestInventory and set inventory to 0 before editing the donation, I will try that. 

I am thinking of including info on the inventory system in CONTRIBUTING.md so new contributors have a easier time writing tests.

Reply:
resolved by [ba5af77](https://github.com/rubyforgood/human-essentials/pull/4483/commits/ba5af77c029d0c303104692f78628b27afa3d220)